### PR TITLE
[MLU] add mlu kernel for reduce_max_grad

### PR DESCRIPTION
**Pull request by Ashelly1**
### PR types

New features
### PR changes

OPs
### Describe

add mlu kernel for reduce_max_grad

### CODE REVIEW

File: paddle/fluid/operators/reduce_ops/reduce_max_op_mlu.cc [hunk 1]
```diff
@@ -92,6 +92,111 @@ class ReduceMaxMLUKernel : public framework::OpKernel<T> {
...
+                         transformed_out_grad_desc.get(),
+                         GetBasePtr(&transformed_out_grad));
+
+    // compare
+    Tensor equal_cond;
+    equal_cond.mutable_data<bool>(x_grad->dims(), place);
+
+    MLUCnnlTensorDesc x_desc(*x);
+    MLUCnnlTensorDesc equal_cond_desc(equal_cond);
+    MLUCnnl::CycleOp(context,
```
Comment (ShawnNew) [inline paddle/fluid/operators/reduce_ops/reduce_max_op_mlu.cc]: 可以尝试使用MLUCnnl::Logic

**Review by ShawnNew — CHANGES_REQUESTED** [review]

File: paddle/fluid/operators/reduce_ops/reduce_max_op_mlu.cc (hunk 1 above)
Comment (Ashelly1) [reply paddle/fluid/operators/reduce_ops/reduce_max_op_mlu.cc]: 已修改

**Review by ShawnNew — DISMISSED** [review]
👍

**Review by ronny1996 — APPROVED** [review]
LGTM

**Review by qili93 — APPROVED** [review]
LGTM

**Pull request merged**